tools/oesgx/oesgx.c

```diff
@@ -162,15 +163,24 @@ int main(int argc, const char* argv[])
             dump_regs(&regs);
             return result;
         }
+
         if (!HAVE_EPC_SUBLEAF(regs))
         {
             printf("No EPC section\n");
             dump_regs(&regs);
             return 0;
         }
-        epc_size =
-            ((regs.ecx & 0x0fffff000) |
-             ((uint64_t)(regs.edx & 0x0fffff) << 32));
+
+        while(!result && HAVE_EPC_SUBLEAF(regs))
+        {
+            epc_size +=
+                ((regs.ecx & 0x0fffff000) |
+                 ((uint64_t)(regs.edx & 0x0fffff) << 32));
+
+            ecx++;
+            regs.ecx = ecx;
+            result = _CPUID(&regs);
```

Comment:
We need to let the user know in case this _CPUID call fails.

Reply:
The _CPUID() call in the line 159 is being reported to the user. The _CPUID() call in line 182 is to check if ecx = 0x3 (or higher) correspond to a subleaf. Is there a way to find out the number of subleafs?

Reply:
Can the call in line 182 fail?

Reply:
yes, and the failure is the exit condition from the while loop..

Reply:
Added comments in src.